misk/src/test/kotlin/misk/hibernate/ManualCharacterQuery.kt

```diff
@@ -0,0 +1,165 @@
+package misk.hibernate
+
+import misk.hibernate.TransacterTest.ActorAndReleaseDate
+import misk.hibernate.TransacterTest.CharacterQuery
+import misk.hibernate.TransacterTest.MovieNameAndReleaseDate
+import java.time.LocalDate
+import javax.inject.Singleton
+import javax.persistence.criteria.CompoundSelection
+import javax.persistence.criteria.CriteriaBuilder
+import javax.persistence.criteria.Path
+import javax.persistence.criteria.Predicate
+import javax.persistence.criteria.Root
+import kotlin.reflect.KClass
+
+// TODO(jwilson): build this with metaprogramming (reflection/codegen) instead of code.
```

Comment:
we must go deeper :)

Reply:
I wonder if we could do something cool and use the indexes of the schema to generate allowable where clauses.

Reply:
we can, but it’s an unbounded set

Reply:
i'm thinking we separate filter building into 2 steps, indexed and unindexed. you must pick an available index filter and then can add an additional unbounded set of unindexed filters.

we could even promote the usage of covered indices by combining indexed fields with a projection.